src/config.ts

```diff
@@ -130,8 +131,8 @@ export async function resolveConnections(config: Configuration, resolver?: strin
       id,
       client,
       connection: {
-        host: (connection as any).host,
-        database: (connection as any).database
+        host: (connection as Knex.ConnectionConfig).host,
```

Comment:
Still doesn't cut it in my opinion. Why doesn't connection have a concrete type?

Reply:
What is the type for `connections` can you check?

Reply:
We shouldn't be doing `as` as much as possible.

Reply:
The type for connection is `string | StaticConnectionConfig | ConnectionConfigProvider`. The string for the connection string, `StaticConnectionConfig` is the type for various db connection config ORed together and `ConnectionConfigProvider` basically returns `StaticConnectionConfig` using both sync and async function.

The problem here is with these multiple types for the single key, we have to force the compiler to take in the type we are using.